packages/indexer-agent/src/network.ts

```diff
@@ -120,16 +133,134 @@ export class Network {
+            gasLimit: Math.ceil(txConfig.gasLimit),
+          }
+          tx = await this.wallet.sendTransaction(transaction)
+        }
+
+        logger.info(`Transaction pending`, { tx: tx })
+
+        const receipt = await this.ethereum.waitForTransaction(
+          tx?.hash,
+          3,
+          this.gasBumpTimeLimit,
+        )
+
+        if (receipt.status == 0) {
+          if (receipt.cumulativeGasUsed.toNumber() > txConfig.gasLimit * 0.9) {
+            throw indexerError(IndexerErrorCode.IE050)
+          }
+          throw indexerError(IndexerErrorCode.IE051)
+        }
+
+        logger.info(`Transaction successfully included in block`, {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          tx: tx?.hash,
+          receipt: receipt,
+        })
+        output = receipt
+        pending = false
+      } catch (error) {
+        txConfig = await this.updateTransactionConfig(logger, txConfig, error)
+        continue
+      }
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return output!
+  }
+
+  async updateTransactionConfig(
+    logger: Logger,
+    txConfig: Record<string, number>,
+    error: Error | IndexerError,
+  ): Promise<Record<string, number>> {
+    if (error instanceof IndexerError) {
+      if (error.code == 'IE050') {
+        txConfig.gasLimit = txConfig.gasLimit * txConfig.gasBump
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        txConfig.nonce! += 1
+      } else if (error.code == 'IE051') {
+        throw indexerError(IndexerErrorCode.IE051)
+      }
+    } else if (error instanceof Error) {
+      if (
+        error.message.includes(
+          'Transaction with the same hash was already imported',
+        )
+      ) {
+        txConfig.nonceOffset += 1
+        txConfig.nonce =
+          (await this.wallet.getTransactionCount('pending')) -
+          txConfig.nonceOffset +
+          1
+      } else if (
+        error.message.includes(
+          'Transaction nonce is too low. Try incrementing the nonce.',
+        )
+      ) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        txConfig.nonce! += 1
+      } else if (
+        error.message.includes('Try increasing the fee') ||
+        error.message.includes(
+          'Transaction gas price supplied is too low. There is another transaction with same nonce in the queue. Try increasing the gas price or incrementing the nonce.',
+        ) ||
+        error.message?.includes('timeout exceeded')
+      ) {
+        if (txConfig.gasPrice == this.gasPriceMax) {
+          throw indexerError(IndexerErrorCode.IE052)
+        } else {
+          txConfig.gasPrice =
+            txConfig.gasPrice * txConfig.gasBump > this.gasPriceMax
+              ? this.gasPriceMax
+              : txConfig.gasPrice * txConfig.gasBump
+        }
+      }
+    }
+    logger.warning('Error sending transaction, retrying', {
```

Comment:
We've been fairly consistent beginning all error messages with `Failed to`, which helps searching the logs. Could we change this to `Failed to send transaction, retrying`?